src/Auth0.Core/Exceptions/RateLimitApiException.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Auth0.Core.Http;
+using System;
+
+namespace Auth0.Core.Exceptions
+{
+    /// <summary>
+    /// Represents errors caused by rate limits being exceeded when making API calls.
+    /// </summary>
+    public class RateLimitApiException : ApiException
+    {
+        public RateLimit RateLimit { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RateLimitApiException"/> class.
+        /// </summary>
+        public RateLimitApiException() {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RateLimitApiException"/> class with a specified error message.
+        /// </summary>
+        /// <param name="message">The error message that explains the reason for the exception.</param>
+        public RateLimitApiException(string message) : base(message)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RateLimitApiException"/> class with a specified error message
+        /// and a reference to the inner exception that is the cause of this exception.
+        /// </summary>
+        /// <param name="message">The error message that explains the reason for the exception.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception, or a null 
+        /// reference if no inner exception is specified.</param>
+        public RateLimitApiException(string message, Exception innerException)
```

Comment:
What kind of inner exception would there be for a rate limit hit?

Reply:
None that I can think of.  You just have to have these constructors for Exceptions in .NET. It's part of the implied contract.